public/marked_entity_files.php

```diff
@@ -83,6 +104,26 @@
                 <td><?php echo escape($row->title); ?></td>
                 <td><?php echo escape($row->description); ?></td>
                 <td><?php echo $row->attachment->file_filename; ?></td>
+                <td>
+                    <?php
+                    $found = false;
+                    foreach ($row->permissions as $permission) {
+                        if ($permission->user_id == $_SESSION["current_user"]->id) {
+                            echo $permission->stringify();
+                            $found = true;
+                            break;
+                        }
+                    }
+                    // Defaults to read for admins, TAs and instructors
+                    if (!$found) {
+                        if (get_current_role() != "student") {
+                            echo "Read";
+                        } else {
+                            echo "N/A";
```

Comment:
the entire blocks looks a little ugly not sure why but I don't have anything better in mind